ksql-rest-app/src/main/java/io/confluent/ksql/rest/entity/ConnectorDescription.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"; you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.rest.entity;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.util.List;
+import java.util.Objects;
+import org.apache.kafka.connect.runtime.rest.entities.ConnectorStateInfo;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class ConnectorDescription extends KsqlEntity {
+
+  private final String type;
+  private final ConnectorStateInfo status;
+  private final List<SourceDescription> sources;
+
+  @JsonCreator
+  public ConnectorDescription(
+      @JsonProperty("statementText")  final String statementText,
+      @JsonProperty("type")           final String type,
+      @JsonProperty("status")         final ConnectorStateInfo status,
+      @JsonProperty("sources")        final List<SourceDescription> sources
+  ) {
+    super(statementText);
+    this.type = Objects.requireNonNull(type, "type");
+    this.status = Objects.requireNonNull(status, "status");
+    this.sources = Objects.requireNonNull(sources, "sources");
+  }
+
+  public String getType() {
+    return type;
+  }
+
+  public ConnectorStateInfo getStatus() {
+    return status;
+  }
+
+  public List<SourceDescription> getSources() {
+    return sources;
+  }
+
+  @Override
+  public boolean equals(final Object o) {
+    if (this == o) {
```

Comment:
It feels like type might be part of the identity for this class. Any reason to leave it out?

Reply:
There is no good reason! I added it after I generated hashcode and equals and forgot to update it 🙂